frontend/src/bundles/video_editor/components/timeline/subcomponents/timeline-view.tsx

```diff
@@ -0,0 +1,101 @@
+import { format, hoursToMilliseconds, minutesToMilliseconds, secondsToMilliseconds } from 'date-fns';
+import  { type ItemDefinition, type RowDefinition } from 'dnd-timeline';
+import { groupItemsToSubrows, useTimelineContext } from 'dnd-timeline';
+import { useMemo } from 'react';
+
+import { Item } from './item.js';
+import { Row } from './row.js';
+import { Subrow } from './subrow.js';
+import { type MarkerDefinition } from './timeaxis.js';
+import { TimeAxis } from './timeaxis.js';
+import { TimeCursor } from './timecursor.js';
+
+const timeAxisMarkers: MarkerDefinition[] = [
+    {
+        value: hoursToMilliseconds(24),
+    },
+    {
+        value: hoursToMilliseconds(2),
+        minRangeSize: hoursToMilliseconds(24),
+    },
+    {
+        value: hoursToMilliseconds(1),
+        minRangeSize: hoursToMilliseconds(24),
+    },
+    {
+        value: hoursToMilliseconds(1),
+        maxRangeSize: hoursToMilliseconds(24),
+        getLabel: (date: Date) => format(date, 'k'),
+    },
+    {
+        value: minutesToMilliseconds(30),
+        maxRangeSize: hoursToMilliseconds(24),
+        minRangeSize: hoursToMilliseconds(12),
+        getLabel: (date: Date) => format(date, 'm'),
+    },
+    {
+        value: minutesToMilliseconds(15),
+        maxRangeSize: hoursToMilliseconds(12),
+        getLabel: (date: Date) => format(date, 'm'),
+    },
+    {
+        value: minutesToMilliseconds(5),
+        maxRangeSize: hoursToMilliseconds(6),
+        minRangeSize: hoursToMilliseconds(3),
+        getLabel: (date: Date) => format(date, 'm'),
+    },
+    {
+        value: minutesToMilliseconds(5),
+        maxRangeSize: hoursToMilliseconds(3),
+        minRangeSize: hoursToMilliseconds(1),
+        getLabel: (date: Date) => format(date, 'm'),
+    },
+    {
+        value: secondsToMilliseconds(30),
+        maxRangeSize: minutesToMilliseconds(30),
+        minRangeSize: minutesToMilliseconds(10),
+        getLabel: (date: Date) => format(date, 'mm:ss'),
+    },
+    {
+        value: 1000,
+        maxRangeSize: minutesToMilliseconds(10),
+        minRangeSize: minutesToMilliseconds(1),
+        getLabel: (date: Date) => format(date, 's'),
+    },
+];
+
+interface TimelineProperties {
+    rows: RowDefinition[];
+    items: ItemDefinition[];
+}
+
+const TimelineView = (properties: TimelineProperties): JSX.Element => {
```

Comment:
here

Reply:
Changed to destructuring approach, thank you.